### PR TITLE
Use more descriptive placeholders

### LINKED ISSUE #10: Use more descriptive place holders
How about using the following place holders, but still support the old ones for backward compatibility.

{user} instead of %u
{reporter} instead of %r
{handler} instead of %h
{project} instead of %p

### PR DESCRIPTION
**Pull request by vboctor**
Fixes #10

### CODE REVIEW

File: Snippets/Snippets.API.php [hunk 1]
```diff
@@ -334,7 +339,11 @@ public static function delete_by_user_id($user_id) {
 	private static function from_db_result($result) {
 		$snippets = array();
 		while ($row = db_fetch_array($result)) {
-			$snippet = new Snippet($row["type"], $row["name"], $row["value"], $row["user_id"]);
+			$snippet = new Snippet(
+                $row['type'],
+                $row['name'],
+                Snippet::replace_legacy_placeholders($row['value']),
+                $row['user_id']);
```
Comment (dregad) [inline Snippets/Snippets.API.php]: Incorrect indentation (spaces instead of tabs ?)

File: Snippets/Snippets.API.php [hunk 2]
```diff
@@ -343,6 +352,21 @@ private static function from_db_result($result) {
...
 
+    /**
+     * Replace legacy placeholders (e.g. %u) with modern ones (e.g. {user}).
+     *
+     * @param string $p_value The snippet to process.
+     * @return string The processed snippet.
+     */
+    private static function replace_legacy_placeholders($p_value) {
+        $t_value = $p_value;
+        $t_value = str_replace('%u', PLACEHOLDER_USER,     $t_value);
```
Comment (dregad) [inline Snippets/Snippets.API.php]: Save one assignment:
``` php
$t_value = str_replace('%u', PLACEHOLDER_USER,     $p_value);
```

File: Snippets/Snippets.API.php [hunk 3]
```diff
@@ -343,6 +352,21 @@ private static function from_db_result($result) {
...
+        $t_value = str_replace('%r', PLACEHOLDER_REPORTER, $t_value);
+        $t_value = str_replace('%h', PLACEHOLDER_HANDLER,  $t_value);
+        $t_value = str_replace('%p', PLACEHOLDER_PROJECT,  $t_value);
+        return $t_value;        
+    }
+
```
Comment (dregad) [inline Snippets/Snippets.API.php]: spaces instead of tabs

**Comment by dregad** [comment]
Looks good

**Comment by dregad** [comment]
Suggest you bump the version number to 1.1.0 after this.

**Comment by vboctor** [comment]
All feedback applied except "saving an assignment".  I find this more readable and less like to introduce errors if line ordering is to change.

**Pull request merged**